### ISSUE TITLE
Playback reverse crash

### ISSUE THREAD

**Issue by netExtra** [mpv-player/mpv#12424]
### Important Information

When reversing playback on mpv build 12.08.2023  up to the latest build on Windows 10, mpv crashes out. Build 11.08.2023 is the last version to allow this. I tried with both vo=gpu and vo=gpu-next.  I also check both the mpv and ffmpeg commits on the 11.08 and 12.08 but nothing stood out as being the obvious cause. 

Specific mpv versions are in the two log files provided. "FAIL mpv.log" is mpv build 12.08.2023 where playback reverse doesn't work. Both mpv binaries are sourced from https://github.com/shinchiro/mpv-winbuild-cmake. The same issue occurs with the latest build.

### Reproduction steps

Both attempts are run with an empty mpv.conf except for : 
log-file=mpv.log
msg-level=all=trace

All scripts, input and config files have been removed except for ffrew-keybinds.lua which I used to run the test with commands:
mp.command("set play-dir +") "Forward"
mp.command("set play-dir -") "Reverse"

### Expected behavior
Video plays in reverse.

### Actual behavior
Video immediately quits.

### Log file
[mpv.log](https://github.com/mpv-player/mpv/files/12653261/mpv.log)
[FAIL mpv.log](https://github.com/mpv-player/mpv/files/12653205/FAIL.mpv.log)

Log files are attached but here is a short excerpt from both log files:

Reverse working:
[   8.971][d][cplayer] Run command: script-binding, flags=73, args=[name="ffrew_keybinds/stepBack"]
[   8.971][t][cplayer] video_output_image: r=2/eof=0/st=playing
[   8.971][t][mkv] bytes=3979360, read_more=0 prefetch_more=0, refresh_more=0
[   8.971][d][cplayer] Run command: set, flags=73, args=[name="play-dir", value="-"]
[   8.971][v][cplayer] Set property: play-dir=- -> 1
[   8.971][t][cplayer] video_output_image: r=2/eof=0/st=playing
[   8.971][t][mkv] bytes=3979360, read_more=0 prefetch_more=0, refresh_more=0
[   8.971][v][mkv] queuing seek to 8.592000
[   8.972][v][mkv] execute seek (to 8.592000 flags 36)
[   8.972][d][ao/wasapi] Thread Reset
[   8.972][t][file] seek request from 24626026 to 42134561
[   8.972][v][file] stream level seek from 24626026 to 42134561
[   8.972][v][mkv] seek done

Reverse FAIL:
[  12.069][d][cplayer] Run command: script-binding, flags=73, args=[name="ffrew_keybinds/stepBack"]
[  12.069][t][cplayer] video_output_image: r=2/eof=0/st=playing
[  12.069][t][mkv] bytes=3972176, read_more=0 prefetch_more=0, refresh_more=0
[  12.069][t][cplayer] frametime=0.042
[  12.069][d][cplayer] Run command: set, flags=73, args=[name="play-dir", value="-"]
[  12.069][v][cplayer] Set property: play-dir=- -> 1
[  12.069][t][osc] rendering 
[  12.070][t][cplayer] Run command: show-text, flags=64, args=[text="Reverse", duration="750", level="0"]
[  12.070][t][mkv] bytes=3864672, read_more=0 prefetch_more=0, refresh_more=0
[  12.070][t][input] release all
[  12.070][t][cplayer] video_output_image: r=2/eof=0/st=playing
[  12.070][t][mkv] bytes=3864672, read_more=0 prefetch_more=0, refresh_more=0
[  12.070][v][mkv] queuing seek to -11.720000
[  12.070][t][ffmpeg/video] hevc: nal_unit_type: 0(TRAIL_N), nuh_layer_id: 0, temporal_id: 0
[  12.070][t][ffmpeg/video] hevc: Output frame with POC 298.
[  12.070][v][mkv] execute seek (to -11.720000 flags 36)
[  12.070][d][ao/wasapi] Thread Reset
[  12.070][t][file] seek request from 33407227 to 62293
[  12.070][v][file] stream level seek from 33407227 to 62293
[  12.070][v][mkv] seek done

**Comment by llyyr** [comment]
Can you check if reverting https://github.com/mpv-player/mpv/commit/692ccca3a2eb48810eb1a8abb346be06a9b17d53 fixes the issue?

**Comment by Dudemanguy** [comment]
Well that's almost certainly the cause but I don't see any crashing in this log so not sure what's supposed to be going wrong in the first place.

**Comment by netExtra** [comment]
> Can you check if reverting [692ccca](https://github.com/mpv-player/mpv/commit/692ccca3a2eb48810eb1a8abb346be06a9b17d53) fixes the issue?
Yep. t was only 6 lines of code so I just removed them and recompiled. Reverse playback works 👍

**Comment by po5** [comment]
What's happening is that mpv now jumps from the current position when setting `play-dir`, to the end of the file for backward, and to the start of the file for forward.  
I don't understand why this was changed. Broke my [evafast](https://github.com/po5/evafast/tree/rewrite) script.

**Comment by Dudemanguy** [comment]
Sorry I forgot about this. Give me a second.

**Comment by po5** [comment]
Can confirm that fixed it, thanks.

**Comment by netExtra** [comment]
> What's happening is that mpv now jumps from the current position when setting `play-dir`, to the end of the file for backward, and to the start of the file for forward. I don't understand why this was changed. Broke my [evafast](https://github.com/po5/evafast/tree/rewrite) script's hold-to-rewind feature.

This is a better description of the behaviour I saw when I opened this ticket. Anyway it no longer exhibits this behaviour so it looks like it is working properly now. I'm having issues but there are always issues with reverse so I will consider this closed. Thanks.
